distributed/client.py

```diff
@@ -2552,7 +2552,86 @@ def _graph_to_futures(
         fifo_timeout=0,
         actors=None,
     ):
+        user_priority = user_priority or {}
+        restrictions = restrictions or {}
+        loose_restrictions = loose_restrictions or []
+        retries = retries or {}
+        resources = resources or {}
+
         with self._refcount_lock:
+            annotation_retries = {}
+            annotation_restrictions = {}
+            annotation_loose_restrictions = []
+            annotation_resources = {}
+
+            if isinstance(dsk, HighLevelGraph):
+                for layer in dsk.layers.values():
+
+                    if layer.annotations is None:
+                        continue
+
+                    layer_priority = layer.annotations.get("priority", False)
+
+                    if layer_priority is not False:
+                        if isinstance(layer_priority, Number):
```

Comment:
FWIW I found using `if "priority" in layer.annotations:` more clear.  There is a bit less to follow.

I think that in general I optimize highly for adding as few terms as possible to code, perhaps more than most (you can probably see this in my previous comments.  This mostly comes from having to debug a high volume of code.  Every time someone introduces a new class or function or even variable that I need to learn it adds significantly to the time that it takes to quickly understand a piece of foreign code.  

I think that spelling everything out, like you do here, makes sense as we write code (it's nice to give everything a name and be explicit) but that it ends up hurting future reviewers who are in a rush.  I write more about this topic here: https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection

Reply:
Yep, no need to optimise per layer.

I've implemented this in the associated PR https://github.com/dask/dask/pull/6889/commits/e0e1dbf86e18a2fe2038fa470045cdea1f4fd155 where checks are performed on the existing distributed Client.{submit, persist, compute} keyword taxonomy.